authors/apps/articles/views.py

```diff
@@ -0,0 +1,204 @@
+from django.shortcuts import render
+from rest_framework import generics, permissions, status, views, viewsets
+from rest_framework.response import Response
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.decorators import permission_classes
+from .serializers import ArticleSerializer, TABLE
+from django.contrib.auth.models import User
+from ..authentication.models import User
+from .models import ArticleModel
+from rest_framework.permissions import (
+    IsAuthenticatedOrReadOnly, IsAuthenticated, IsAdminUser, AllowAny
+)
+from django.http import JsonResponse
+from django.core.files.storage import FileSystemStorage
+import json
+from django.http.response import Http404
+import smtplib
+from django.conf import settings
+from .utils import ImageUploader
+from django.apps import apps
+
+
+class ArticleView(viewsets.ModelViewSet):
+    """
+    The article View
+    """
+
+    queryset = ArticleModel.objects.all()
+    serializer_class = ArticleSerializer
+
+    permission_classes_by_action = {
+        'create': [IsAuthenticated],
+        'list': [AllowAny],
+        'retrieve': [AllowAny],
+        'update': [IsAuthenticated],
+        'destroy': [IsAuthenticated],
+    }
+
+    def get_permissions(self):
+        """
+        Function to handle permissions
+        """
+        try:
+            return [permission() for permission in self.permission_classes_by_action[self.action]]
+        except KeyError:
+            return [permission() for permission in self.permission_classes]
+
+
+    def create(self, request):
```

Comment:
You can look to validate for duplicate articles where a user should not be able to add the same article twice.

Reply:
Done